src/msca/metrics/main.py

```diff
@@ -0,0 +1,254 @@
+        obs : str
+            Column name for observed/actual values
+        pred : str
+            Column name for predicted values
+        weights : str
+            Column name for sample weights
+        groupby : list[str], optional
+            Column names to group by for grouped calculations
+
+        Returns
+        -------
+        Union[float, pd.DataFrame]
+            Single metric value if no groupby, DataFrame with grouped results if groupby specified
+        """
+        if groupby is not None:
+            return self._eval_grouped(data, obs, pred, weights, groupby)
+
+        return self._eval_single(data, obs, pred, weights).iloc[0]
+
+    def eval_skill(
+        self,
+        data: pd.DataFrame,
+        obs: str,
+        pred_alt: str,
+        pred_ref: str,
+        weights: str,
+        groupby: list[str] | None = None,
+    ) -> float | pd.DataFrame:
+        """
+        Calculate skill score by comparing pred_alt performance against pred_ref.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            Input DataFrame containing all required columns
+        obs : str
+            Column name for observed/actual values
+        pred_alt : str
+            Column name for alternative predicted values to evaluate
+        pred_ref : str
+            Column name for reference predicted values to compare against
+        weights : str
+            Column name for sample weights
+        groupby : list[str], optional
+            Column names to group by for grouped calculations
+
+        Returns
+        -------
+        Union[float, pd.DataFrame]
+            Single skill score if no groupby, DataFrame with grouped skill scores if groupby specified
+        """
+        if groupby is not None:
+            ref_scores = self._eval_grouped(
+                data=data,
+                obs=obs,
+                pred=pred_ref,
+                weights=weights,
+                groupby=groupby,
+            )
+            alt_scores = self._eval_grouped(
+                data=data,
+                obs=obs,
+                pred=pred_alt,
+                weights=weights,
+                groupby=groupby,
+            )
+
+            ref_score_col = f"{pred_ref}_{self.value}"
+            alt_score_col = f"{pred_alt}_{self.value}"
+            result_column_name = f"{pred_alt}_{self.value}_skill"
+
+            if (ref_scores[ref_score_col] == 0).any():
+                zero_ref_groups = ref_scores[ref_scores[ref_score_col] == 0][
+                    groupby
+                ].to_dict("records")
+                raise ZeroDivisionError(
+                    f"Reference score is zero for groups {zero_ref_groups}, cannot calculate skill score"
+                )
+
+            grouped_results = ref_scores.copy()
+            grouped_results[result_column_name] = 1.0 - (
+                alt_scores[alt_score_col] / ref_scores[ref_score_col]
+            )
```

Comment:
nit: we can visit this later, I will feel safer if we did a merge between `ref_scores` and `alt_scores` by `groupby`, so that we don't have to worry about row mismatch. But since both of them are generated with the same `DataFrame` and same function they should be the same. But merge is safer.